src/sequelize.js

```diff
@@ -333,6 +277,22 @@ export class Sequelize {
       dialectOptions: this.options.dialectOptions,
     };
 
+    // Convert replication connection strings to objects
+    if (this.options.replication) {
+      if (this.options.replication.write && typeof this.options.replication.write === 'string') {
+        this.options.replication.write = this.parseConnectionString(this.options.replication.write);
+      }
+
+      if (this.options.replication.read && this.options.replication.read.length) {
```

Comment:
redundant check, the for loop will not execute if the length of the array is 0:

```suggestion
      if (this.options.replication.read) {
```